src/cache_refresh/cache_refresh.cc

```diff
@@ -91,14 +100,21 @@ int refreshpasswdcache() {
   }
   cache_file.close();
 
-  if (count > 0) {
-    if (rename(kDefaultBackupFilePath, kDefaultFilePath) != 0) {
-      syslog(LOG_ERR, "Could not move passwd cache file.");
+  if (error_code == ENOMSG) {
+    remove(kDefaultBackupFilePath);
+    return 0;
+  } else if (error_code == ENOENT) {
+    syslog(LOG_ERR, "Failed to get users, not updating passwd cache file, removing %s.", kDefaultBackupFilePath);
+    // If the cache file already exists, don't re-write it if we get a server
+    // error. Else, an empty cache file will be written below.
+    if (fexists(kDefaultFilePath)) {
```

Comment:
i dont understand - if the real file exists, then remove the temporary file? how does it match the intention in the comment?

Reply:
If the real file exists, we don't want to over-write it with the current temporary file on a server error. So we should remove the temporary file and return here, so we don't execute the code after which tries to move the temporary file to the real file. I've updated the comment, I hope that makes it less confusing.

Reply:
i missed that this block ends in 'return', it makes sense now